### PR TITLE
chore(FX-3568): Search Cleanup

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Chore**

This PR resolves [FX-3568]

### Acceptance criteria:
- [x] Get rid of SearchSwitchContainer
- [x] Remove the old Search implementation and rename the Search2 into Search
- [x] Keep the components that we were using from Search directory in Eigen in place
- [x] Remove improved search feature flag (if we can and there is no problem) (from `features.ts`)
- [x] Move SearchInput to its own file

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Search Cleanup after release - dimatretyak

</details>

[FX-3568]: https://artsyproduct.atlassian.net/browse/FX-3568?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/lib/Scenes/Search/Search.tsx [hunk 1]
```diff
@@ -1,68 +1,302 @@
...
+interface TappedSearchResultData {
+  query: string
+  type: string
+  position: number
+  contextModule: ContextModule
+  slug: string
+  objectTab?: string
+}
+
+const SearchInputContainer = connectSearchBox(SearchInput)
```
Comment (gkartalis) [inline src/lib/Scenes/Search/Search.tsx:49]: **suggestion(non-blocking):** Should we move this to be in the same file as the search input and import it from there? What do you think?

File: src/lib/Scenes/Search/Search.tsx [hunk 2]
```diff
@@ -1,68 +1,302 @@
...
+const SearchResultsContainerWithState = connectStateResults(SearchResults)
+const SearchResultsContainer = connectInfiniteHits(SearchResultsContainerWithState)
```
Comment (gkartalis) [inline src/lib/Scenes/Search/Search.tsx:51]: same with this should we move this to the SearchResults component and import it from there? 🤔

**Review by gkartalis — APPROVED** [review]

File: src/lib/Scenes/Search/Search.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/lib/Scenes/Search/Search.tsx:49]: I tried to keep the `SearchInput` component "silly". And store all the connection logic in `Search`

File: src/lib/Scenes/Search/Search.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/lib/Scenes/Search/Search.tsx:51]: Here is the same situation as above

File: src/lib/Scenes/Search/Search.tsx (hunk 1 above)
Comment (gkartalis) [reply src/lib/Scenes/Search/Search.tsx:49]: sounds good. No problem with this approach 👍  (same goes to the next one! ) Feel free to merge whenever you want

**Review by tr-ann — APPROVED** [review]
LGTM 💯

**Pull request merged**